depends/Makefile

```diff
@@ -125,6 +125,10 @@ $(host_arch)_$(host_os)_id_string+=$(shell $(host_CXX) --version 2>/dev/null)
 $(host_arch)_$(host_os)_id_string+=$(shell $(host_RANLIB) --version 2>/dev/null)
 $(host_arch)_$(host_os)_id_string+=$(shell $(host_STRIP) --version 2>/dev/null)
 
+ifneq ($(strip $(FORCE_USE_SYSTEM_CLANG)),)
+$(host_arch)_$(host_os)_id_string+=system_clang
```

Comment:
Sorry, I realize this was my commit..

In testing, I noticed that we need to add this to the build string too. Because for darwin builds it uses the weird trick of compiling the native builds with the downloaded compiler, we need to redo those builds when we go back and forth between system-clang and non.

```diff
diff --git a/depends/Makefile b/depends/Makefile
index b1d929d8dd..2bc5df974a 100644
--- a/depends/Makefile
+++ b/depends/Makefile
@@ -126,6 +126,7 @@ $(host_arch)_$(host_os)_id_string+=$(shell $(host_RANLIB) --version 2>/dev/null)
 $(host_arch)_$(host_os)_id_string+=$(shell $(host_STRIP) --version 2>/dev/null)

 ifneq ($(strip $(FORCE_USE_SYSTEM_CLANG)),)
+build_id_string+=system_clang
 $(host_arch)_$(host_os)_id_string+=system_clang
 endif
```

Reply:
Added to: 156b604203ef17b2b77ee9dacf15e375c809242a